timemachine/fe/free_energy.py

```diff
@@ -154,11 +155,35 @@ class InitialState:
     lamb: float
     ligand_idxs: NDArray
     protein_idxs: NDArray
+    atom_map: Optional[AtomMapMixin] = None
```

Comment:
Add a comment that this is needs to be set for RBFE only (since we also use initial states to run MD)

Reply:
Comment added in https://github.com/proteneer/timemachine/pull/1338/commits/0b0a8921ad7b0d35875243facf083c8914ac78e5

Reply:
(independently, I've been wondering recently whether it's worth adding `AtomMapMixin` to `SimulationResult`, since I end up reconstructing the mapping in a lot investigations)

Reply:
>  I've been wondering recently whether it's worth adding AtomMapMixin to SimulationResult

Part of the reason I added it to the InitialState, is that I end up having to reconstruct the AtomMapMixin a lot, but I tend to work with InitialStates rather than SimulationResults. 

Will play around with constructing the interacting atoms in the `setup_initial_state` function. If we do that do we want to break pickling by adding in a positional argument?

Reply:
Changed to interacting atoms in https://github.com/proteneer/timemachine/pull/1338/commits/0202c84561001738705484397b716d884c3a95c6

Didn't change `setup_initial_states` in `absolute_hydration.py` since it looks unused